int32     message_code
int32[]   ids
string[]  ip_addresses
int32[]   ports
float64[] timestamps
string[]  roles
string[]  statuses